### PR TITLE
cleanup container runtime options

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove the deprecated flags in v1.27

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
please correct me if not

#### Does this PR introduce a user-facing change?

```release-note
kubelet: remove deprecated flag `--container-runtime`
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/config/flags.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/config/flags.go:58]: Unfortunately this is not implemented yet on containerd: https://github.com/containerd/containerd/issues/6352

Perhaps instead of removing it, extend it for couple more releases

/hold

Comment (pacoxu) [reply pkg/kubelet/config/flags.go:58]: Line 55 removing seems to be valid.

@SergeyKanzhelev should @calvin0327 keep `pod-infra-container-image` and remove the `container-runtime` in this PR?

Comment (SergeyKanzhelev) [reply pkg/kubelet/config/flags.go:58]: Yes

Comment (calvin0327) [reply pkg/kubelet/config/flags.go:58]: thanks, I have recovered the `pod-infra-container-image`

Comment (pacoxu) [reply pkg/kubelet/config/flags.go:58]: /unhold

File: pkg/kubelet/config/flags.go
Comment (pacoxu) [inline pkg/kubelet/config/flags.go:54]: This should be removed along with `ContainerRuntime` related code.

Comment (calvin0327) [reply pkg/kubelet/config/flags.go]: done

**Comment by sftim** [comment]
Does this PR need a release note?

File: pkg/kubelet/config/flags.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/config/flags.go:55]: can you please update the description of `pod-infra-container-image` in this same PR. So there will be no follow up PRs attempting to remove it

**Comment by bart0sh** [comment]
@calvin0327  If you still need this PR then please rebase, if not, please close the PR.

**Comment by marquiz** [comment]
ping @calvin0327 are you still working on this? If so, please rebase or then close

**Comment by pacoxu** [comment]
> ping @calvin0327 are you still working on this? If so, please rebase or then close

rebased on behalf of @calvin0327

**Comment by pacoxu** [comment]
/release-note-edit
```release-note
kubelet: remove deprecated flag `--container-runtime`
```

File: cmd/kubelet/app/options/options.go
Comment (marquiz) [inline cmd/kubelet/app/options/options.go:182]: Hmm, adding this new code for checking about seccompdefault seems totally unrelated to dropping the container-runtime option

File: pkg/kubelet/config/flags.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/config/flags.go:54]: there are still a few mentions of `--container-runtime` in codebase, for example in utils.sh and local-up.sh. Can you please confirm they are still needed and add comment there why or clean up. I remember last time we tried to remove those some of the tests started failing

Comment (pacoxu) [reply pkg/kubelet/config/flags.go:54]: Find another 3 places and I updated the PR.
1. cmd/kubelet/app/options/options.go: `This flag can only be used with --container-runtime=remote.`
2. cluster/gce/util.sh
3. hack/local-up-cluster.sh

**Comment by marquiz** [comment]
Thanks @pacoxu for the update
/lgtm

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
we have lgtm from @SergeyKanzhelev and it looks good to me as well.

/approve
/lgtm

**Pull request merged**
